src/components/Editable.tsx

```diff
@@ -249,7 +249,7 @@ const Editable = ({
 
   // side effect to set old value ref to head value from updated simplePath.
   useEffect(() => {
-    oldValueRef.current = value
+    oldValueRef.current = isEditing ? removeMarkdownFormatting(value) : formatMarkdown(value)
```

Comment:
There is no need to use `oldValueRef`, as we can just convert the markdown when rendering the ContentEditable. I will push a commit showing this.

Reply:
Your point is genuine. The current approach is forcing bidirectional conversion inside the `ContentEditable`. I've updated the code to work without updating the `oldValueRef` and applied unidirectional conversion inside of `ContentEditable`. But there was a condition that would reset the `innerHTML` to whatever is there in the `oldValueRef`, possibly to disallow redundant state updates. I've updated the condition to compare the `oldValueRef.current` with `removeMarkdownFormatting(contentRef.current?.innerHTML)`. I also fixed the not re-rendering issue by tracking previous value of `isEditing` inside a new ref.